core/iwasm/libraries/lib-socket/test/nslookup.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2019 Intel Corporation.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <assert.h>
+#include <string.h>
+#ifdef __wasi__
+#include <wasi_socket_ext.h>
+#endif
+#include <netdb.h>
+
+void
+testNsLookup(int af)
+{
+
+    struct addrinfo *res;
```

Comment:
even though it's a test, I think we should attempt to free the structure as well (e.g. if we ever run it with sanitizers, it'll give us some hints in case something goes wrong). Use `freeaddrinfo` for that.